SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -174,25 +184,24 @@ function Set-Secret
         $Keys = $Secret.Keys.Where({$_ -notin $SpecialKeys })
         foreach ($k in $Keys) {
             if ($Secret.$k -is [securestring]) {
-                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.$k)
-                $Secret.$k = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.$k = [System.Net.NetworkCredential]::new("", $Secret.$k).Password
             }
             $sb.AppendLine("$($k): $($Secret.$k)") | Out-Null
         }
 
         # Notes need to be on a new line
         if ($null -ne $Secret.Notes) {
             if ($Secret.Notes -is [securestring]) {
-                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.Notes)
-                $Secret.Notes = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.Notes = [System.Net.NetworkCredential]::new("", $Secret.Notes).Password
             }
             $sb.AppendLine("Notes: `n$($Secret.Notes)") | Out-Null
         }
     } 
     
     try {
-        $res = Invoke-lpass 'show', '--sync=now', '--name', $Name, '2>/dev/null' -ErrorAction Stop 
-        $SecretExists = $null -ne $res 
+        $res = Invoke-lpass 'show', '--sync=now', '--name', $Name
+        # We use ToString() here to turn the ErrorRecord into a string if we got an ErrorRecord
+        $SecretExists = $res.ToString() -ne "Error: Could not find specified account(s)."
```

Comment:
I think I would add a null check just in case. 
What do you think ?
```suggestion
        $SecretExists = $null -ne $res -and $res.ToString() -ne "Error: Could not find specified account(s)."
```